synapse/federation/sender/per_destination_queue.py

```diff
@@ -415,22 +416,93 @@ async def _catch_up_transmission_loop(self) -> None:
                     "This should not happen." % event_ids
                 )
 
-            if logger.isEnabledFor(logging.INFO):
-                rooms = [p.room_id for p in catchup_pdus]
-                logger.info("Catching up rooms to %s: %r", self._destination, rooms)
+            # We send transactions with events from one room only, as its likely
+            # that the remote will have to do additional processing, which may
+            # take some time. It's better to give it small amounts of work
+            # rather than risk the request timing out and repeatedly being
+            # retried, and not making any progress.
+            for pdu in catchup_pdus:
```

Comment:
Should we be grouping these by room first? Otherwise it seems like we would do some unnecessary work. (I think at least...)